receiver/libhoneyreceiver/receiver.go

```diff
@@ -191,10 +191,14 @@ func (r *libhoneyReceiver) handleEvent(resp http.ResponseWriter, req *http.Reque
 
 	body, err := io.ReadAll(req.Body)
 	if err != nil {
+		r.settings.Logger.Error("Failed to read request body", zap.Error(err))
 		errorutil.HTTPError(resp, err)
+		return
```

Comment:
Should we return these errors and let the caller of `handleEvent` handle them? Should they eventually be returned to the client?

Reply:
Good point. this isn't how errors should be dealt with at all. 

`handleEvent()` is a fire and forget function here so the http response body/code for the HTTP request here is done in the handleEvent() function. I did notice that by using the `errorutil.HTTPError()`, it isn't obeying the msgpack/json encoding or libhoney response type so I'm switching it to the proper response for libhoney clients.